sys/linux/socket.txt

```diff
@@ -161,6 +177,24 @@ cmsghdr {
 	data		array[int8]
 } [align_ptr]
 
+type cmsghdr_t[SOL, TYP, DATA] {
+	unaligned	cmsghdr_unaligned[SOL, TYP, DATA]
+} [align_ptr]
+
+type cmsghdr_unaligned[SOL, TYP, DATA] {
+	cmsg_len	len[parent, intptr]
+	cmsg_level	const[SOL, int32]
+	cmsg_type	const[TYP, int32]
+	data		DATA
+} [packed]
+
+# For __sock_cmsg_send
+cmsghdr_sock [
+	mark		cmsghdr_unaligned[SOL_SOCKET, SO_MARK, int32]
```

Comment:
This also should use cmsghdr_t because each msg in the array needs to be aligned to intptr.